src/controller/python/test/test_scripts/custom_commissioning_test.py

```diff
@@ -0,0 +1,185 @@
+    optParser.add_option(
+        "-t",
+        "--timeout",
+        action="store",
+        dest="testTimeout",
+        default=75,
+        type='int',
+        help="The program will return with timeout after specified seconds.",
+        metavar="<timeout-second>",
+    )
+    optParser.add_option(
+        "--bad-cert-issuer",
+        action="store_true",
+        dest="badCertIssuer",
+        default=False,
+        help="Simulate a bad certificate issuer, the commissioning should fail when sending OpCreds.",
+    )
+    optParser.add_option(
+        "-a",
+        "--address",
+        action="store",
+        dest="deviceAddress",
+        default='',
+        type='str',
+        help="Address of the device",
+        metavar="<device-addr>",
+    )
+    optParser.add_option(
+        "--setup-payload",
+        action="store",
+        dest="setupPayload",
+        default='',
+        type='str',
+        help="Setup Payload (manual pairing code or QR code content)",
+        metavar="<setup-payload>"
+    )
+    optParser.add_option(
+        "--nodeid",
+        action="store",
+        dest="nodeid",
+        default=1,
+        type=int,
+        help="The Node ID issued to the device",
+        metavar="<nodeid>"
+    )
+    optParser.add_option(
+        '--paa-trust-store-path',
+        dest="paaPath",
+        default='',
+        type='str',
+        help="Path that contains valid and trusted PAA Root Certificates."
+    )
+
+    (options, remainingArgs) = optParser.parse_args(sys.argv[1:])
+
+    timeoutTicker = TestTimeout(options.testTimeout)
+    timeoutTicker.start()
+
+    test = BaseTestHelper(
+        nodeid=112233, paaTrustStorePath=options.paaPath, testCommissioner=True)
+
+    class BadCredentialProvider:
+        def __init__(self, devCtrl: ChipDeviceCtrl.ChipDeviceController):
+            self._devCtrl = devCtrl
+
+        async def get_attestation_nonce(self) -> bytes:
+            return os.urandom(32)
+
+        async def get_csr_nonce(self) -> bytes:
+            return os.urandom(32)
+
+        async def get_commissionee_credentials(self, request: commissioning.GetCommissioneeCredentialsRequest) -> commissioning.GetCommissioneeCredentialsResponse:
+            node_id = random.randint(100000, 999999)
+            nocChain = self._devCtrl.IssueNOCChain(Clusters.OperationalCredentials.Commands.CSRResponse(
+                NOCSRElements=request.csr_elements, attestationSignature=request.attestation_signature), nodeId=node_id)
+            return commissioning.GetCommissioneeCredentialsResponse(
+                rcac=nocChain.rcacBytes[1:],
+                noc=nocChain.nocBytes[1:],
+                icac=nocChain.icacBytes[1:],
+                ipk=nocChain.ipkBytes[1:],
+                case_admin_node=self._devCtrl.nodeId,
+                admin_vendor_id=self._devCtrl.fabricAdmin.vendorId,
+                node_id=node_id,
+                fabric_id=self._devCtrl.fabricId)
+
```

Comment:
Real example implementations of this would determine the fabric ID, compressed fabric ID and node ID from the NOC subject, not the devCtrl.

We need to make the building blocks that:

- GetCommissioneeCredentialsResponse that returns the commissionee credentials in DER format (not Matter cert format) --> what is the output of `New*X509Cert` as in ExampleOperationalCredentialsIssuer.
- Conversion routine from DER to Matter format (calls `ConvertX509CertToChipCert`) as in `GenerateNOCChainAfterValidation`

Our cloud services generate DER certs. Python code will get those. We then need to convert them to matter cert

Reply:
We can do this in our own implementation, and we can provide a helper function to translate the DER / PEM to TLV